File: runtime/ruby/units/rt/i64_rem_s.rb
# requires: rt/s64, rt/trap
def i64_rem_s(a, b)
  sa = s64(a)
  sb = s64(b)
  trap("integer divide by zero") if sb == 0
  r = sa.abs % sb.abs
  r = -r if sa < 0
  r & M64
end

File: runtime/python/units/rt/i64_rem_s.py
# requires: rt/s64, rt/trap
@staticmethod
def i64_rem_s(a, b):
    sa = Rt.s64(a)
    sb = Rt.s64(b)
    if sb == 0:
        Rt.trap("integer divide by zero")
    r = abs(sa) % abs(sb)
    if sa < 0:
        r = -r
    return r & Rt.M64
